vassal-app/src/main/java/VASSAL/build/module/WizardSupport.java

```diff
@@ -519,20 +519,76 @@ public Component getListCellRendererComponent(JList<?> list, Object value, int i
       return box;
     }
 
+    // From 3.7.6, load Pre-defined setups in the Foreground, not in the Background as arrowing through the PDS drop-down can initiate
+    // multiple concurrent game loads for any but the smallest save files.
     protected void loadSetup(PredefinedSetup setup, final WizardController controller, final Map settings) {
+      controller.setProblem(Resources.getString("WizardSupport.LoadingGame"));
       final GameModule g = GameModule.getGameModule();
+
+      // First check the specified PDS actually exists in the module
+      final String fileName = setup.getFileName();
       try {
-        new SavedGameLoader(controller, settings, new BufferedInputStream(setup.getSavedGameContents()), POST_PLAY_OFFLINE_WIZARD, true).start();
+        if (!g.getDataArchive().contains(fileName)) {
+          controller.setProblem(Resources.getString("WizardSupport.UnableToLoad"));
+          return;
+        }
       }
-      // FIXME: review error message
-      catch (IOException e1) {
+      catch (Exception e) {
         controller.setProblem(Resources.getString("WizardSupport.UnableToLoad"));
+        return;
       }
-      // Predefined Scenarios are files too...
+
+      try {
+        final BufferedInputStream bis = new BufferedInputStream(setup.getSavedGameContents());
+        Command setupCommand = GameModule.getGameModule().getGameState().decodeSavedGame(bis);
+        bis.close();
+        try {
+          if (setupCommand == null) {
+            throw new IOException(Resources.getString("WizardSupport.InvalidSavefile")); //$NON-NLS-1$
+          }
+          // Strip out the setup(true) command. This will be applied when the "Finish" button is pressed
+          setupCommand = new CommandFilter() {
+            @Override
+            protected boolean accept(Command c) {
+              return !(c instanceof GameState.SetupCommand) ||
+                !((GameState.SetupCommand) c).isGameStarting();
+            }
+          }.apply(setupCommand);
+        }
+        catch (IllegalStateException e) {
+          final String msg = e.getMessage();
+          if (msg != null && msg.startsWith(Resources.getString("Decorator.no_state_for_trait"))) { //BR//
+            WarningDialog.show("GameState.probably_wrong_version"); //NON-NLS
+          }
+          else {
+            throw e;
+          }
+        }
+
+        setupCommand.execute();
```

Comment:
That's by design. Previously, the IllegalStateException was not caught and just generated a bug/error dialog in all cases and the save game could not be loaded or recovered. 

We tracked down a subset of these as being caused by loading out of date saved games where the state recorded for some counters no longer matched the definition of the counters in the game. If this case is determined, then we now show them a warning and then try and load the save game, (it may only affect one counter) and give them the chance to retrieve the game. The state of some counters may be incorrect, but on the whole, the save game is generally usable.